### PR TITLE
fix(matches_related_rule): Add check to match Kind and Resource for m…

### PR DESCRIPTION
**Pull request by newtondev**
Fixes https://github.com/metacontroller/metacontroller/issues/883

### CODE REVIEW

**Comment by newtondev** [comment]
Going to cleanup the tests with a test table.

**Comment by newtondev** [comment]
@grzesuav I have add the fix and the tests, ready for review when you have a chance, thanks.

File: pkg/controller/common/customize/manager_test.go [hunk 1]
```diff
@@ -218,3 +218,315 @@ func TestDetermineSelectionType_returnLabelsWhenNothingIsPresent(t *testing.T) {
...
+				Namespace:     "some",
+				Names:         []string{"name"},
+			},
+			wantMatch: false,
+		},
+		// When parent is cluster scoped
+		{
+			name:         "return true when name and namespace matches",
+			isNamespaced: false,
+			parent:       fakeGenericParentWithNamespace(),
```
Comment (grzesuav) [inline pkg/controller/common/customize/manager_test.go:369]: @newtondev I believe here parent should not be namespaced?

File: pkg/controller/common/customize/manager_test.go [hunk 2]
```diff
@@ -218,3 +218,315 @@ func TestDetermineSelectionType_returnLabelsWhenNothingIsPresent(t *testing.T) {
...
+					APIVersion: "some",
+					Resource:   "some",
+				},
+				LabelSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"aaa": "bbb"}},
+			},
+			wantMatch: true,
+		},
+		{
+			name:   "return false when labels do not match",
+			parent: fakeGenericParent(),
```
Comment (grzesuav) [inline pkg/controller/common/customize/manager_test.go:266]: parent should be namespaced here

File: pkg/controller/common/customize/manager_test.go (hunk 1 above)
Comment (grzesuav) [reply pkg/controller/common/customize/manager_test.go:369]: and in the following

**Review by grzesuav — APPROVED** [review]

**Comment by newtondev** [comment]
Thanks @grzesuav , very much appreciated.

**Pull request merged**